### DISCUSSION TITLE
Detecting a full queue from a publisher object

### DISCUSSION THREAD

**Discussion by forestf255** [eclipse-iceoryx/iceoryx#2032] [Q&A] [answered]
Does Iceoryx support either returning an error to the publisher or otherwise detecting when a subscribers queue is full? Neither the [QoS policies documentation](https://iceoryx.io/latest/concepts/qos-policies/#compatible-policies) nor the [BasePublisher](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/include/iceoryx_posh/internal/popo/base_publisher.hpp) API indicate this is possible, but wanted to make sure I am not missing anything. It seems like detecting overwritten/lost data is only visible from the [subscriber](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/include/iceoryx_posh/internal/popo/base_subscriber.hpp#L100)

**Comment by elBoberido** [comment, marked as answer]
@forestf255 there is no API for that. The problem is that a publisher can have multiple subscriber and not all of the queues might be in the same state

**Reply by elfenpiff** [reply]
@forestf255 We provide a publisher option to wait when a subscriber queue is full, see: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_examples/iceoptions/iox_publisher_with_options.cpp

Maybe this helps you to solve your problem.
